### PR TITLE
L2-322: Stake neurons from subaccounts

### PR DESCRIPTION
**Pull request by lmuntanerDfinity**
# Motivation

User can stake a neuron from subaccounts.

# Changes

* Add subaccounts to SelectAccount screen in CreatenNeuronModal.
* Add `fromSubAccount` parameter to all the stake neuron flow.
* New helper to transform from SubAccountArray to number that is needed for nns-js.

# Tests

* New test to check that user can select a subaccount in CreatenNeuronModal.

### CODE REVIEW

**Comment by lmuntanerDfinity** [comment]
@mstrasinskis @peterpeterparker please review

File: frontend/svelte/src/lib/api/utils.api.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import type { SubAccountArray } from "../canisters/nns-dapp/nns-dapp.types";
+
+// Source: as in ts/src/canisters/converters.ts
+export const arrayOfNumberToUint8Array = (
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/api/utils.api.ts:4]: all these functions are exposed by nns-js

**Review by peterpeterparker — CHANGES_REQUESTED** [review]

File: frontend/svelte/src/lib/api/utils.api.ts (hunk 1 above)
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/api/utils.api.ts:4]: They might, but they are related to nns dapp canister, not to the ledger or governance canisters. Therefore, I decided to move them here and pass as subaccountIt what the ledger canister needs, not the type SubAccountArray, which is a type from NNS Dapp canister.

Does that make sense?

Or we also want nns dapp specific code in nns-js?

Comment (peterpeterparker) [reply frontend/svelte/src/lib/api/utils.api.ts:4]: There functions are used within nns-js (toNeuron -> arrayOfNumberToUint8Array). Therefore I would not duplicate code.

Comment (peterpeterparker) [reply frontend/svelte/src/lib/api/utils.api.ts:4]: But if you find a way to not use the functions within nns-js, I am not against moving the code in nns-dapp. Just don't like duplicate code ;)

Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/api/utils.api.ts]: Removed unused code in nns-js in https://github.com/dfinity/nns-js/pull/62

And also refactored here.

**Comment by lmuntanerDfinity** [comment]
@peterpeterparker ready for another review

**Review by peterpeterparker — APPROVED** [review]
LGTM

Since you have move `arrayBufferToNumber` and `toSubAccountId` it's maybe a good opportunity to add some tests for these? Not necessary in this PR.

File: frontend/svelte/src/lib/api/utils.api.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/api/utils.api.ts]: All clear, thx

**Comment by lmuntanerDfinity** [comment]
Yes, I thought about tests for those, but to be honest, I would be copying the implementation in the test for the expected value.

I am not sure how that could help. Or just hardcode the current expected value and put it in a test. At least if we change the implementation we would be aware, I will do that in another PR.

**Pull request merged**